rust_src/src/alloc.rs

```diff
@@ -57,4 +65,39 @@ pub fn bool_vector(args: &mut [LispObject]) -> LispObject {
     vector
 }
 
+/// Create a new record.
+/// TYPE is its type as returned by `type-of'; it should be either a
+/// symbol or a type descriptor.  SLOTS is the number of non-type slots,
+/// each initialized to INIT.
+#[lisp_fn]
+pub fn make_record(r#type: LispObject, slots: EmacsUint, init: LispObject) -> LispObject {
```

Comment:
That '#' is a typo right?

Reply:
No, it's a raw identifier to preserve the original argument name, but if that's not necessary then I'll change it.

Reply:
There are a few other instances where a `type` arg was renamed as `itype` or something like that. We should fix those with this technique.

Reply:
Remember to update the docstring when you rename an argument.

And kudos, I learned something about Rust today.

Reply:
It's a relatively recent addition if I remember correctly, I think it got added to stable within the last 6 months or so.